warehouse/static/js/warehouse/controllers/notification_controller.js

```diff
@@ -51,5 +55,9 @@ export default class extends Controller {
       localStorage.setItem(notificationId, 1);
     }
     this.notificationTarget.classList.remove("notification-bar--visible");
+    if (this.isSticky) {
+      let bodyElement = document.querySelector("body");
+      bodyElement.style.paddingTop = "0px";
+    }
```

Comment:
Can you explain why this is necessary? This PR seems to solve the issue fine without the changes here, and I don't think we really want to be adding style attributes in the stimulus controllers.

Reply:
without this when dismissing the warning banner the `padding-top` on the body will stay at the previous value, effectively not collapsing

![preprod_warning_collapse](https://user-images.githubusercontent.com/6739793/38240270-16bdb7d6-3727-11e8-97a5-ef696f04daa0.gif)

I agree it's not very elegant to change the `body` style from the Stimulus controller, but given the dismissal logic lives there I couldn't think of a better way to collapse the body on dismissal.

Reply:
Is this possibly related to https://github.com/pypa/warehouse/issues/3519 ?

Reply:
@nlhkabu Yes, seems likely.